components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/prompt/GeckoPromptDelegate.kt

```diff
@@ -413,18 +411,21 @@ internal class GeckoPromptDelegate(private val geckoEngineSession: GeckoEngineSe
 
     private fun Uri.toFileUri(context: Context): Uri {
         val uri = this
-        val temporalFile = java.io.File(context.cacheDir, uri.getFileName(context))
-        try {
-            val inStream = context.contentResolver.openInputStream(uri) as FileInputStream
-            val outStream = FileOutputStream(temporalFile)
-            val inChannel = inStream.channel
-            val outChannel = outStream.channel
-            inChannel.transferTo(0, inChannel.size(), outChannel)
-            inStream.close()
-            outStream.close()
-        } catch (e: IOException) {
-            val logger = Logger("GeckoPromptDelegate")
-            logger.warn("Could not convert uri to file uri", e)
+        val cacheUploadDirectory = java.io.File(context.cacheDir, "/uploads")
+
+        if (!cacheUploadDirectory.exists()) {
+            cacheUploadDirectory.mkdir()
+        }
+
+        val temporalFile = java.io.File(cacheUploadDirectory, uri.getFileName(context))
+        (context.contentResolver.openInputStream(uri) as FileInputStream).use { inStream ->
+            inStream.use {
```

Comment:
Seems like `inStream` is in use for the second time.
I suppose only two `use` declarations requires (as we have two streams).
Also there still need to catch exception in case if [it'll be thrown](https://github.com/JetBrains/kotlin/blob/master/libraries/stdlib/jvm/src/kotlin/io/Closeable.kt#L27).

I thought about something like:
```
try {
    inStream.use {
        outStream.use {
		    // ...
        }
    }
} catch (/*...*/) {
    // ...
}
```

Reply:
Sorry, I pushed by mistake, I'm update it locally.

Reply:
Updated :) !